lix/source-code/client/src/helpers/withLazyFetching.ts

```diff
@@ -0,0 +1,189 @@
+import type { NodeishFilesystem } from '@lix-js/fs';
+import { checkout, listFiles } from 'isomorphic-git';
+
+import { blobExistsLocaly } from "./blobExistsLocaly.js"
+import { fetchBlobsFromRemoteThrottled } from "./fetchBlobsFromRemoteThrottled.js"
+// import { fetchBlobsFromRemote } from "./fetchBlobsFromRemote.js"
+// eslint-disable-next-line unicorn/prefer-node-protocol -- TODO #1459 we explicitly want to use path to use browserfy-path here
+import path from "path"
+
+/**
+ * Wraps a nodeishFs implementation with a js proxy for detailed logging, debugging and transparently replacing the file access behaviour.
+ * advantage of using this approach is that the underlying fs can also be swapped to something like lightingfs seamlessly.
+ */
+export const withLazyFetching = (
+	fs: NodeishFilesystem,
+	dir: string,
+	gitdir: string,
+	ref: string,
```

Comment:
ref, dir, gitdir, oidToFilePaths not required for first version, lets simplify args